schnorr_fun/src/frost.rs

```diff
@@ -892,38 +920,36 @@ impl<H: Digest<OutputSize = U32> + Clone, NG> Frost<H, NG> {
         s!(r1 + (r2 * b) + lambda * x * c).public()
     }
 
-    /// Verify a partial signature for a participant at `index` (from zero).
-    ///
-    /// ## Return Value
-    ///
-    /// Returns `bool`, true if partial signature is valid.
+    /// Verify a partial signature for a participant at `index`
     pub fn verify_signature_share(
         &self,
         frost_key: &FrostKey<EvenY>,
         session: &SignSession,
         index: PartyIndex,
         signature_share: Scalar<Public, Zero>,
-    ) -> bool {
+    ) -> Result<(), VerifyShareError> {
         let s = signature_share;
-        let mut lambda = lagrange_lambda(
+        let mut lambda = eval_lagrange_basis_polynomial(
+            Scalar::<Public, _>::zero(),
             index,
             session.nonces.keys().filter(|&j| *j != index).copied(),
         );
         lambda.conditional_negate(frost_key.needs_negation);
         let c = &session.challenge;
         let b = &session.binding_coeff;
-        let X = match frost_key.verification_shares().get(&index) {
-            Some(key) => key,
-            None => return false,
-        };
-        let [R1, R2] = session
+        let X = frost_key.verification_share(index);
+        let [ref R1, ref R2] = session
             .nonces
             .get(&index)
-            .expect("verifying party index that is not part of frost signing coalition")
+            .ok_or(VerifyShareError::PartyIsNotPartyOfSession)?
             .0;
         let R1 = R1.conditional_negate(session.nonces_need_negation);
         let R2 = R2.conditional_negate(session.nonces_need_negation);
-        g!(R1 + b * R2 + (c * lambda) * X - s * G).is_zero()
+        if g!(R1 + b * R2 + (c * lambda) * X - s * G).is_zero() {
+            Ok(())
+        } else {
+            Err(VerifyShareError::InvalidShare)
```

Comment:
This should be `InvalidSignatureShare`.

Reply:
Agreed! Maybe we will rename SignatureShare -> PartialSignature https://github.com/LLFourn/secp256kfun/issues/155